lib/at_monitor/at_monitor.c

```diff
@@ -79,10 +92,9 @@ static void at_monitor_task(struct k_work *work)
 
 	while ((at_notif = k_fifo_get(&at_monitor_fifo, K_NO_WAIT))) {
 		/* Match notification with all monitors */
-		LOG_DBG("AT notif: %s", log_strdup(at_notif->data));
+		LOG_DBG("AT notif: %.*s", strlen(at_notif->data) - strlen("\r\n"), at_notif->data);
 		STRUCT_SECTION_FOREACH(at_monitor_entry, e) {
-			if (!e->paused &&
-			   (e->filter == ANY || strstr(at_notif->data, e->filter))) {
+			if (!is_paused(e) && !is_direct(e) && has_match(e, at_notif->data)) {
```

Comment:
This will now loop over a section containing both direct and non-direct monitors, so a check for `is_direct()` is necessary.